firstvoices/backend/resources/pages.py

```diff
@@ -0,0 +1,73 @@
+import logging
+
+from import_export import fields
+from jwt_auth.models import User
+
+from backend.models import Image, Site, SitePage
+from backend.models.constants import Visibility
+from backend.models.media import Video
+from backend.models.widget import SiteWidget, SiteWidgetList, SiteWidgetListOrder
+from backend.resources.base import SiteContentResource
+from backend.resources.utils.import_export_widgets import ChoicesWidget
+
+
+class SitePageResource(SiteContentResource):
+    visibility = fields.Field(
+        column_name="visibility",
+        widget=ChoicesWidget(Visibility.choices),
+        attribute="visibility",
+    )
+
+    class Meta:
+        model = SitePage
+
+    @staticmethod
+    def get_user_or_none(email):
+        try:
+            return User.objects.get(email=email)
+        except User.DoesNotExist:
+            return None
+
+    def before_import_row(self, row, **kwargs):
+        logger = logging.getLogger(__name__)
+
+        # Create a SiteWidgetList object for each SitePage and convert the list of widget IDs to SiteWidgetListOrder
+        # objects connecting the list to the widgets.
+        if row["widgets"] == "":
+            row["widgets"] = None
+        else:
+            last_modified_by_user = self.get_user_or_none(row["last_modified_by"])
+            created_by_user = self.get_user_or_none(row["created_by"])
+            site = Site.objects.get(id=row["site"])
+            widgets_list = row["widgets"].split(",")
+            site_widget_list = SiteWidgetList.objects.create(site=site)
```

Comment:
Just wondering if these object creation steps could potentially cause an error or some orphans if we imported with `dry_run=True`? Or should we just commit to avoid that?

Reply:
I believe if we run with `dry_run=True` almost every import will fail since they all depend on some other table (mainly the sites). I think we can probably leave this as is and not use `dry_run=True` but I am open to other suggestions.